EIPS/eip-969.md

```diff
@@ -0,0 +1,167 @@
+`fnvB` replaces `fnv` in the DAG item mix step
+`fnvC(fnvD(fnvE` replaces `fnv(fnv(fnv(` in the compress mix step
+
+`fnv` as utilized in DAG-item creation should remain unchanged.
+
+## Agent Changes (Optional Variant)
+ 
+GetWork may optionally return the proposed blocks algorithm. While a 
+miner or pool may infer the requirement for ethashV2 based on the computed 
+epoch of the provided seedHash, it is beneficial to explicitly provide this
+field so a miner does not require special configuration when mining on a chain
+that chooses not to implement the ASIC_Mitigation hardfork.
+
+Due to compatibility concerns with implementations that already add additional 
+parameters to GetWork, it is desired to add BlockNumber as an explicit 4th parameter 
+as suggested https://github.com/ethereum/go-ethereum/issues/2333. Algorithm 
+should then be returned as either `"ethash"` or `"ethashV2"` according to the 
+`block.number >= ASIC_MITIGATION_FORK_BLKNUM` criteria. 
+  
+## Rationale
+
+This EIP is aimed at breaking existing ASIC based miners via small changes to the
+existing ethash algorithm.  We hope to accomplish the following:
+
+1. Break existing ASIC based miners.
+2. Demonstrate a willingness to fork in the event of future ASIC miner production.
+
+Goal #1 is something that we can only do probabalistically without detailed
+knowledge of existing ASIC miner design.  Our approach should balance the
+inherent security risks involved with changing the mining algorithm with the
+risk that the change we make does not break existing ASIC miners.  This EIP
+leans towards minimizing the security risks by making minimal changes to the
+algorithm accepting the risk that the change may not break dedicated hardware 
+miners that utilize partially or fully configurable logic. 
+
+Furthermore, we do not wish to introduce significant algorithm changes that
+may alter the power utilization or performance profile of existing GPU hardware.
+
+The change of FNV constant is a minimal change that can be quickly
+implemented across the various network node and miner implementations.
+
+It is proposed that `ASIC_MITIGATION_FORK_BLKNUM` be no more than 5550000 (epoch 185), giving
+around 30 days of notice to node and miner developers and a sufficient window
+for formal analysis of the changes by experts. We must weigh this window against
+the risk introduced by allowing ASICs that may exist to continue to propagate
+on the network, as well as the risk of providing too much advanced warning to 
+ASIC developers. 
+
+It is further understood that this change will not prevent redesign of existing
+dedicated hardware with new ASIC chips. The intention of this change is only
+to disable currently active or mid-production hardware and provide time for
+POS development as well as larger algorithim changes to be well analyzed by 
+experts.
+
+The choice of FNV constants is made based on the formal specification at
+https://tools.ietf.org/html/draft-eastlake-fnv-14#section-2.1
```

Comment:
That's good, but it would be good to be a bit more explicit, e.g. as per the comment I linked to above.

Reply:
Similarly for this link please change it to `specification [here](https://tools.ietf.org/html/draft-eastlake-fnv-14#section-2.1)` or `specification at [https://tools.ietf.org/html/draft-eastlake-fnv-14#section-2.1](https://tools.ietf.org/html/draft-eastlake-fnv-14#section-2.1).

Reply:
The proposed goals are too conservative and risky. I afraid the network could just pay for a hard fork and get nothing:
1- I disagree with the argument that the nature of Bitmain attack is a mystery, it is based on Dager's vulnerability to shared memory attack inherited by Ethash and not metigated properly, let's be more rational. 
2- The commitment to not to 'alter the power utilization or performance profile of existing GPU hardware' is totally useless: first of all any tweak will alter both indexes somehow and secondly there is no cause not to alter each of them because it won't cause any loses as long as it is fairly distributed among are legitimate architectures. 
My recommendation: Be more aggressive.

Reply:
@aliasrafd do you have any evidence to support your claim? What we know about the E3 does not such they have used any such shares memory scheme (in fact such schemes are generally not practical in reality, you need GH rates before there is any practical chance of reusing DAG items fetched from memory for great gain, below that you need just as much memory bandwidth to store your in-progress hashes. Furthermore existing memory is already maxing out its bus, so no more transactions could be supported), and their performance envelope is well within the capabilities of a normal architecture. 

Regarding the desire not to change the performance or power characteristics, that could easily be relaxed if there was a compelling reason to do so. This change is intentionally minimal. Additionally the index changes would not affect this profile, as you are simple changing from one random distribution to another.

Reply:
@AirSquirrels Is there anything you guys know about E3 that I don't? Please share if any. Otherwise, the most important fact (besides it is about Bitmain, a beast but not a scammer)  is that the device has exactly the same efficiency as a gpu configuration, isn't it? So, it is basically a gpu based system but a lot more cost effective. Putting this together with the inherent Dagger vulnerability to shared memory attack, you come to what conclusion? 
Do you think bitmain has invented a super cheap gpu as efficient as AMD and NVIDIA gpus but a lot more cheaper? How? By eliminating some useless parts of the architecture or adding some cryptographic circuits to it or both? Got any evidence for such claims? Has Bitmain any history relevant for taking such a huge steps forward. This company is not even a manufacturer of its own ASIC chips! 
As of your arguments about memory bandwidth bottleneck and your claimed memory bandwidth being 'maxed out': It is not really 'maxed out' that much, An Nvidia Titan Z  gpu has 5760 Cuda cores and it doesn't maximize its DRAM bandwidth, achieving a linear gain in its total performance. 
Now suppose Bitmain has managed for (buying/building) a high throughput buss controller and DRAM architecture, although not a breakthrough, just state of the art, now sharing this resource with a bunch of cheap gpus, thanks to Dagger vulnerability to such a schema, they can achieve what they are claiming: a more cost effective alternative for usual gpu rigs with almost a same efficiency in terms of Js/H index.
My proposed scenario also is supported by other claimed specifications for E3 like its ability to be 'reprogrammed' what this EIP has explicitly admitted its possibility as a drawback for the proposed tweak.